sdk/servicebus/service-bus/test/internal/2048-limit-test.spec.ts

```diff
@@ -0,0 +1,396 @@
+      ...newArgs: any
+    ) {
+      const link = await _getCurrentReceiver.apply(this, newArgs);
+      setIncomingCapacityOnLink(link, newCapacity);
+      return link;
+    };
+    return batchingReceiver;
+  };
+}
+
+describe("2048 scenarios - receiveBatch in a loop", function(): void {
+  beforeEach(() => {
+    serviceBusClient = createServiceBusClientForTests({
+      retryOptions
+    });
+  });
+
+  afterEach(async () => {
+    await serviceBusClient.test.afterEach();
+    return serviceBusClient.test.after();
+  });
+
+  async function receiveMessages(numberOfMessagesToReceive: number) {
+    let messages: ServiceBusReceivedMessage[] = [];
+    while (messages.length < numberOfMessagesToReceive) {
+      messages = messages.concat(await receiver.receiveMessages(50, { maxWaitTimeInMs: 5000 }));
+    }
+    chai.assert.equal(
+      messages.length,
+      numberOfMessagesToReceive,
+      "Unexpected number of messages received"
+    );
+    return messages;
+  }
+
+  describe("receiveAndDelete", () => {
+    getTestClientTypes().forEach((clientType) => {
+      it(
+        clientType + ": would be able to receive more than bufferCapacity messages",
+        async function(): Promise<void> {
+          await beforeEachTest(clientType, "receiveAndDelete");
+          await sendMessages(numberOfMessagesToSend);
+          await receiveMessages(numberOfMessagesToSend);
+          await verifyMessageCount(0, entityName);
+        }
+      );
+    });
+  });
+
+  describe("peekLock: can receive a max of (bufferCapacity-1) messages when not being settled", () => {
+    getTestClientTypes().forEach((clientType) => {
+      it(
+        clientType +
+          ": deliveryCount will be incremented for (bufferCapacity-1) messages if closed the receiver and received again",
+        async function(): Promise<void> {
+          await beforeEachTest(clientType);
+          await sendMessages(numberOfMessagesToSend);
+          await Promise.all(
+            (await receiveMessages(bufferCapacityToSet - 1)).map((msg) =>
+              receiver.abandonMessage(msg)
+            )
+          );
+          await verifyMessageCount(numberOfMessagesToSend, entityName);
+          await serviceBusClient.close();
+          serviceBusClient = createServiceBusClientForTests();
+          await verifyMessageCount(numberOfMessagesToSend, entityName);
+          receiver = await serviceBusClient.test.createReceiveAndDeleteReceiver(entityName);
+          const messages = await receiveMessages(numberOfMessagesToSend);
+          const delCount = new Array(10).fill(0, 0, 10);
+          for (const message of messages) {
+            if (message.deliveryCount) {
+              delCount[message.deliveryCount]++;
+            }
+          }
+          chai.assert.equal(
+            delCount[1],
+            bufferCapacityToSet - 1,
+            "Unexpected number of messages have deliveryCount = 1"
```

Comment:
Logged issue for unincremented delivery count for sessions https://github.com/Azure/azure-sdk-for-js/issues/15116